### PR TITLE
Feature/ab#79 relation part2

### PR DESCRIPTION
**Pull request by witchpou**
Relationship validation: please review ab#79 first part and ab#49 first.

## Screenshots (if appropriate):
image

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the code style of this project.

### CODE REVIEW

File: webclient/app/src/app/commons/relationshipAccordion/RelationshipAccordion.jsx [hunk 1]
```diff
@@ -62,98 +47,99 @@ function RelationshipAccordion(props) {
...
                             >
-                                {prepareEntitiesSelection()?.map(entityName => (
-                                    <MenuItem value={entityName}>{entityName}</MenuItem>
+                                {targetEntities.map(entity => (
+                                    <MenuItem value={entity.name} key={entity.id} >{entity.name}</MenuItem>
                                 ))}
                             </Select>
+                            <FormHelperText id="targetEntityHelperText"
+                            error={!RegexConfig.entityTitle.test(relationship.otherEntityName)}
+                            >{ t("relationship.targetEntity.description") }</FormHelperText>
```
Comment (DerCed) [inline webclient/app/src/app/commons/relationshipAccordion/RelationshipAccordion.jsx:119]: I do not think you need this here. The label in the select says the same as the Helper text

File: webclient/app/src/app/commons/relationshipAccordion/RelationshipAccordion.jsx [hunk 2]
```diff
@@ -62,98 +47,99 @@ function RelationshipAccordion(props) {
...
+                                label={ t("relationship.sourceField") }
+                                value={relationship.relationshipName}
                                 onChange={(event) => editRelationshipProperty("relationshipName", event.target.value)}
-                                disabled={!currentEntity.fields}
-                            >
-                                {currentEntity?.fields?.map(field => (
-                                    <MenuItem value={field.fieldName}>{field.fieldName}</MenuItem>
-                                ))}
-                            </Select>
+                                helperText={ t("relationship.sourceField.description", {entityName: relationship.otherEntityName ? relationship.otherEntityName : t("relationship.targetEntity")} )}
```
Comment (DerCed) [inline webclient/app/src/app/commons/relationshipAccordion/RelationshipAccordion.jsx:91]: See below

File: webclient/app/src/app/commons/entityCard/Entity.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+interface EntityDataType {
```
Comment (DerCed) [inline webclient/app/src/app/commons/entityCard/Entity.js:1]: I think this structure would be better handled in a separate folder than in entityCard. Maybe a structures folder would be more suitable

File: webclient/app/src/app/commons/entityCard/EntityCard.jsx [hunk 4]
```diff
@@ -24,39 +24,43 @@ function EntityCard(props) {
...
+                        <TableCell>{field.fieldType}</TableCell>
+                        <TableCell />
+                    </TableRow>
+                )
+            })
+        }
     }
 
     function renderFieldsTable(entity) {
-        if (!entity.fields || entity.fields.length === 0) {
```
Comment (DerCed) [inline webclient/app/src/app/commons/entityCard/EntityCard.jsx:45]: Why do you want to print the header, even if there is no content?

File: webclient/app/src/app/commons/entityCard/EntityCard.jsx [hunk 5]
```diff
@@ -24,39 +24,43 @@ function EntityCard(props) {
...
-                    <TableCell className={entityCardStyles.relationshipPlaceholder}>
-                        <div className={"anchor_" + entity.name + "_" + field.fieldName + "_l"}/>
-                    </TableCell>
-                    <TableCell>{field.fieldName}</TableCell>
-                    <TableCell>{field.fieldType}</TableCell>
-                    <TableCell className={entityCardStyles.relationshipPlaceholder}>
-                        <div className={"anchor_" + entity.name + "_" + field.fieldName + "_r"}/>
+                <TableRow key={0}>
+                    <TableCell colSpan={4} >
+                        <div className={entityCardStyles.statementWrapper}>
```
Comment (DerCed) [inline webclient/app/src/app/commons/entityCard/EntityCard.jsx:31]: I do not know, this is best practice to place this inside of an table

File: webclient/app/src/app/commons/entityDialog/EntityDialog.jsx [hunk 6]
```diff
@@ -55,11 +54,34 @@ function EntityDialog(props) {
...
+        });
         setHasFormError(hasError);
 
+
     }, [entity])
 
+
+    function getTargetEntities() {
+        let newTargetEntities = [];
+        if (entities && entities.length > 1) {
```
Comment (DerCed) [inline webclient/app/src/app/commons/entityDialog/EntityDialog.jsx:74]:
```suggestion
        if (entities?.length > 1) {
```
This should work as well

File: webclient/app/src/app/commons/entityDialog/EntityDialog.jsx [hunk 7]
```diff
@@ -101,11 +123,9 @@ function EntityDialog(props) {
         }
     ];
 
-    const relationshipTypes = [
-        "one-to-one",
-        "one-to-many",
-        "many-to-many"
-    ];
+    function lowerFirstChar(s:string) {
```
Comment (DerCed) [inline webclient/app/src/app/commons/entityDialog/EntityDialog.jsx:126]: Is this typescript?

File: webclient/app/src/app/commons/entityDialog/EntityDialog.jsx [hunk 8]
```diff
@@ -260,7 +279,7 @@ function EntityDialog(props) {
     return (
         <Dialog open={!!entityId} maxWidth={"xl"} fullWidth>
             <DialogTitle className={entityEditorStyles.dialogHeaderBar}>
-                <Typography noWrap variant={"h6"} component={"p"}>{t("entityDialog.editDomain")}</Typography>
```
Comment (DerCed) [inline webclient/app/src/app/commons/entityDialog/EntityDialog.jsx:263]: Why did you remove the component tag?

File: webclient/app/src/app/commons/relationshipAccordion/Relationship.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
+const RelationshipType = {
```
Comment (DerCed) [inline webclient/app/src/app/commons/relationshipAccordion/Relationship.js:1]: See structure folder comment above

File: webclient/app/src/app/commons/relationshipAccordion/RelationshipAccordion.jsx [hunk 10]
```diff
@@ -8,49 +8,34 @@ import {
...
 import PropTypes from "prop-types";
-import RelationshipAccordionStyles from "./RelationshipAccordionStyles";
+import RelationshipStyles from "./RelationshipStyles";
 import {useTranslation} from "react-i18next";
+import { RelationshipType } from "./Relationship";
+import ValidatedTextField from "../validatedTextField/ValidatedTextField";
+import RegexConfig from "../../../regexConfig";
 
-
+//some commit
```
Comment (DerCed) [inline webclient/app/src/app/commons/relationshipAccordion/RelationshipAccordion.jsx:23]:
```suggestion

```

File: webclient/app/src/app/commons/relationshipAccordion/RelationshipAccordion.jsx [hunk 11]
```diff
@@ -62,98 +47,99 @@ function RelationshipAccordion(props) {
...
+                            <ValidatedTextField
+                                label={ t("relationship.targetField") }
+                                value={relationship.otherEntityRelationshipName}
                                 onChange={(event) => editRelationshipProperty("otherEntityRelationshipName", event.target.value)}
-                            >
-                                {prepareEntitiesFieldSelection().map(field => (
-                                    <MenuItem value={field.fieldName}>{field.fieldName}</MenuItem>
-                                ))}
-                            </Select>
+                                helperText={ t("relationship.targetField.description", {entityName: currentEntity.name ? currentEntity.name : t("relationship.sourceEntity")} )}
```
Comment (DerCed) [inline webclient/app/src/app/commons/relationshipAccordion/RelationshipAccordion.jsx:127]: See above

**Review by DerCed — CHANGES_REQUESTED** [review]

File: webclient/app/src/app/commons/entityCard/EntityCard.jsx (hunk 5 above)
Comment (witchpou) [reply webclient/app/src/app/commons/entityCard/EntityCard.jsx:31]: I need a places to add the relationship anchors even if there is no field defined

File: webclient/app/src/app/commons/entityCard/Entity.js (hunk 3 above)
Comment (witchpou) [reply webclient/app/src/app/commons/entityCard/Entity.js:1]: the structure doesn't suit at all for me. We can discuss this later.

File: webclient/app/src/app/commons/entityCard/EntityCard.jsx (hunk 4 above)
Comment (witchpou) [reply webclient/app/src/app/commons/entityCard/EntityCard.jsx:45]: I need a places to add the relationship anchors even if there is no field defined

File: webclient/app/src/app/commons/relationshipAccordion/RelationshipAccordion.jsx (hunk 2 above)
Comment (witchpou) [reply webclient/app/src/app/commons/relationshipAccordion/RelationshipAccordion.jsx:89]: helperText was deleted

File: webclient/app/src/app/commons/entityDialog/EntityDialog.jsx [hunk 12]
```diff
@@ -21,7 +21,9 @@ import Statement from "../statement/Statement";
 import {useTranslation} from "react-i18next";
 import ValidatedTextField from "../validatedTextField/ValidatedTextField";
 import RegexConfig from "../../../regexConfig";
+import { defaultRelationship } from "../relationshipAccordion/Relationship";
 import {LoadingButton} from "@mui/lab";
+import { nullEntity } from "../entityCard/Entity";
```
Comment (DerCed) [inline webclient/app/src/app/commons/entityDialog/EntityDialog.jsx:26]: Null Entity?
